wandb/sdk/wandb_init.py

```diff
@@ -1284,15 +1284,27 @@ def try_create_root_dir(settings: Settings) -> None:
             if the parent directory is not writable.
     """
     try:
-        if not os.path.exists(settings.root_dir):
-            os.makedirs(settings.root_dir, exist_ok=True)
+        if os.path.exists(settings.root_dir) and not os.access(
+            settings.root_dir, os.W_OK | os.R_OK
+        ):
+            root_dir = tempfile.gettempdir()
+            wandb.termwarn(
+                f"Path {settings.root_dir} wasn't read/writable, "
+                f"using system temp directory {root_dir}.",
+                repeat=False,
+            )
+            settings.root_dir = root_dir
```

Comment:
We shouldn't duplicate this code. I don't think we need a helper function though; it may be possible to organize the logic here differently to avoid the duplication.